rpcs3/Emu/SysCalls/Modules/cellSysutil.cpp

```diff
@@ -200,23 +202,20 @@ s32 cellSysutilUnregisterCallback(s32 slot)
 	return CELL_OK;
 }
 
-struct CellSysCacheParam
-{
-	char cacheId[CELL_SYSCACHE_ID_SIZE];
-	char getCachePath[CELL_SYSCACHE_PATH_MAX];
-	vm::ptr<void> reserved;
-};
-
 s32 cellSysCacheClear(void)
 {
 	cellSysutil.Todo("cellSysCacheClear()");
 
-	//if some software expects CELL_SYSCACHE_ERROR_NOTMOUNTED we need to check whether
-	//it was mounted before, for that we would need to save the state which I don't know
-	//where to put
+	if (g_sysutil->cacheMounted.exchange(false))
+	{
+		return CELL_SYSCACHE_ERROR_NOTMOUNTED;
+	}
+
 	std::string localPath;
 	Emu.GetVFS().GetDevice(std::string("/dev_hdd1/cache/"), localPath);
 
+	// TODO: Delete everything in the cache folder, except the README
```

Comment:
Seems like I made a small mistake with cacheMounted state, I'll fix and correct the TODO. I'll write some hardware tests in the near future to see what exactly does cellSysCacheClear delete, and what error code it returns when cellSysCacheMount hasn't been called previously.